packages/@aws-cdk/aws-iam/lib/policy-document.ts

```diff
@@ -90,12 +91,16 @@ export class AccountPrincipal extends ArnPrincipal {
  * An IAM principal that represents an AWS service (i.e. sqs.amazonaws.com).
  */
 export class ServicePrincipal extends PolicyPrincipal {
-  constructor(public readonly service: string) {
+  constructor(public readonly service: string, private readonly region?: string) {
     super();
   }
 
   public policyFragment(): PrincipalPolicyFragment {
-    return new PrincipalPolicyFragment({ Service: [ this.service ] });
+    const factName = Facts.servicePrincipal(this.service);
```

Comment:
If this gives me a fact *name* instead of a *fact*, shouldn't the class be called `FactName.servicePrincipal()`?

Reply:
Yeah I guess :)